spec/controllers/internal_api/v1/pillars_controller_spec.rb

```diff
@@ -434,97 +434,107 @@
+  context "with Dex LDAP connectors" do
+    before do |example|
+      # calculate a likely-unique 0-100 integer hash value by adding up the
+      # integer values of chars in the example name (context + example)
+      u = example.metadata[:full_description].codepoints.reduce(0, :+) % 100
+      @conn_name = "LDAP connector #{u}"
+      @host_name = "ldap_host_#{u}.com"
+    end
 
-      expected_json = {
-        registries:          [],
-        kubelet:             {
-          :"compute-resources" => {},
-          :"eviction-hard"     => ""
+    def expected_dex_json(num, certificate, c_name, host)
+      {
+        type:            "ldap",
+        id:              num,
+        name:            c_name,
+        root_ca_data:    Base64.encode64(certificate.certificate),
+        bind:            {
+          anonymous: false,
+          dn:        "cn=admin,dc=#{host.chomp(".com")},dc=com",
+          pw:        "pass"
         },
-        system_certificates: [],
-        dex:                 {
-          connectors: [
-            expected_dex_json(dex_connector_ldap.id, certificate).merge(
-              server:    "ldap_host_#{dex_connector_ldap.id}.com:636",
-              start_tls: false
-            )
-          ]
+        username_prompt: "Username",
+        user:            {
+          base_dn:  "cn=users,dc=#{host.chomp(".com")},dc=com",
+          filter:   "(objectClass=person)",
+          attr_map: {
+            username: "uid",
+            id:       "uid",
+            email:    "mail",
+            name:     "name"
+          }
+        },
+        group:           {
+          base_dn:  "cn=groups,dc=#{host.chomp(".com")},dc=com",
+          filter:   "(objectClass=group)",
+          attr_map: {
+            user:  "uid",
+            group: "member",
+            name:  "name"
+          }
         }
       }
-      get :show do
+    end
+
+    # can't use "let" in a before block where the vars are defined, so need instance vars
+    # rubocop:disable RSpec/ExampleLength, RSpec/InstanceVariable
+    context "with tls" do
+      it "has dex LDAP connectors" do
+        dex_connector_ldap = create(:dex_connector_ldap, :tls, :regular_admin,
+          name: @conn_name, host: @host_name)
+        CertificateService.create(service: dex_connector_ldap, certificate: certificate)
+
+        expected_json = {
+          registries:          [],
+          kubelet:             {
+            :"compute-resources" => {},
+            :"eviction-hard"     => ""
+          },
+          system_certificates: [],
+          dex:                 {
+            connectors: [
+              expected_dex_json(dex_connector_ldap.id, certificate, @conn_name, @host_name).merge(
+                server:    "#{@host_name}:636",
+                start_tls: false
+              )
+            ]
+          }
+        }
+        get :show
         expect(json).to eq(expected_json)
-        delete(dex_connector_ldap)
```

Comment:
This is left in the test database and effects later tests if this line is removed. It should not be removed.